ocaml/idl/datamodel.ml

```diff
@@ -2389,6 +2389,19 @@ let vm_call_plugin = call
 	~allowed_roles:_R_VM_OP
 	()
 
+let vm_import_from = call
+	~name:"import_from"
+	~in_product_since:rel_dundee
+	~doc:"Import an XVA from a URI"
+	~params:[String, "url", "The URL of the XVA file";
```

Comment:
Yeah, perhaps `VM.import` since that name is free (as rob says), plus this
API seems quite general.

On Fri, Jun 12, 2015 at 10:52 AM, Rob Hoes <notifications@github.com> wrote:

> In ocaml/idl/datamodel.ml
> <https://github.com/xapi-project/xen-api/pull/2254#discussion_r32303426>:
>
> > @@ -2389,6 +2389,19 @@ let vm_call_plugin = call
> >  	~allowed_roles:_R_VM_OP
> >  	()
> >
> > +let vm_import_from = call
> > +	~name:"import_from"
> > +	~in_product_since:rel_dundee
> > +	~doc:"Import an XVA from a URI"
> > +	~params:[String, "url", "The URL of the XVA file";
>
> I find import_from a bit vague... Can we make it something like
> import_from_url? Or simple import, since we don't actually have that as
> an API call.
>
> Also, the doc string says URI, but the param has URL.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xapi-project/xen-api/pull/2254/files#r32303426>.
>



-- 
Dave Scott